GoEddie/SQLCover
# working tsqlt tests in github actions for pull requests

related to  #53.

Please fix #59 or add **hacktoberfest-accepted** label to this pr

Changes proposed in this pull request:
 - Got tsqlt tests to run as part of github actions pipeline. The dacpac, create.sql script must be built in **windows**. service containers for running integrated db testing are only supported in **linux** so after building the create sql upload the artifact, spin up a linux machine w/ the sql server and deploy the create.sql then run the tsqlt tests exporting the results to an xml file.
- fixed a sql unit test
- tsqlt pre deploy script for the sp_configure options
- run the non integration c# tests in windows (will be using dotnet core when ported)
- progress towards making sqlcover linux based w/ tsqlt running on linux

How to test this code:
 - see github action or checks tab of this pr

Note:
- github actions have the worst junit report viewers of all of them (vs azure devops and gitlab pipelines), could not get the action to read the junit xml report always giving a 0 out of 0 tests. https://github.com/ashley-taylor/junit-report-annotations-action

Has been tested on (remove any that don't apply):
 - SQL Server 2019


mention @GoEddie
